featurebyte/service/feature_list.py

```diff
@@ -160,6 +165,7 @@ async def _extract_feature_data(self, document: FeatureListModel) -> Dict[str, A
         feature_store_id: Optional[ObjectId] = None
         feature_namespace_ids = set()
         features = []
+        self.feature_list_entity_relationship_validator.reset()
```

Comment:
nit: it is possible to make the validator stateless? Maybe the `validate()` can take a list of relationships info directly, so there is no need for `reset()`.